libpromises/cf3lex.l

```diff
@@ -163,18 +167,28 @@ promise_type   [a-zA-Z_]+:
 
                           if (3 == sscanf(Version(), "%d.%d.%d", &major, &minor, &patch))
                           {
-                            if (2 == sscanf(version_text, "%d.%d", &request_major, &request_minor))
+                            request_level=sscanf(version_text, "%d.%d.%d", &request_major, &request_minor, &request_patch);
+                            if (request_level >= 1)
                             {
-                            if ((request_major > major) ||
-                                (request_major == major && request_minor > minor))
-                              {
-                                /* ignore to the next @endif */
+                              if (request_major > major) {
                                 ParserDebug("\tL:macro @if %d:ignoring to next @endif or EOF\n", P.line_pos);
                                 BEGIN(if_ignore_state);
-                              }
-                              else /* we're OK with the version */
-                              {
+                              } else if (request_major < major || request_level == 1) {
```

Comment:
Won't ``request_level`` always be at least 2? It doesn't look like the ``version_text`` can ever contain less than the major and minor version number, because of the regular expression in ``macro_if_version``.